deltacat/compute/compactor_v2/steps/merge.py

```diff
@@ -306,13 +299,114 @@ def _compact_tables(
         compacted_table=compacted_table,
     )
     total_deduped_records = hb_table_record_count - len(table)
+    return table, incremental_len, total_deduped_records, merge_time
+
+
+def _compact_incremental_and_delete_table(
```

Comment:
As discussed, let's reuse the `build_incremental_table` and `merge_tables` and `apply_deletes` from `DeleteStrategy`.